pkg/storage/streams/store.go

```diff
@@ -279,26 +266,39 @@ func (s *streamStore) Get(ctx context.Context, path paths.Path) (rr ranger.Range
 	if err != nil {
 		return nil, Meta{}, err
 	}
-
 	lastSegmentRanger, lastSegmentMeta, err := s.segments.Get(ctx, encPath.Prepend("l"))
 	if err != nil {
 		return nil, Meta{}, err
 	}
 
-	streamMeta := pb.StreamMeta{}
-	err = proto.Unmarshal(lastSegmentMeta.Data, &streamMeta)
+	derivedKey, err := path.DeriveContentKey(s.rootKey)
```

Comment:
You have this code almost everywhere before you call `getDecryptedStreamInfo()`

```
derivedKey, err := path.DeriveContentKey(s.rootKey)
if err != nil {
	return nil, Meta{}, err
}
key := (*eestream.Key)(derivedKey)
```
I think you should change `getDecryptedStreamInfo` to take in a path and the rootKey, and you move this repeated logic inside that function. That way you can get rid of a lot of extra code

Reply:
yup i can do that!